iam.tf

```diff
@@ -122,7 +122,7 @@ data "aws_iam_policy_document" "logs" {
     effect = "Allow"
 
     actions = compact([
-      !var.use_existing_cloudwatch_log_group ? "logs:CreateLogGroup" : "",
+      !var.use_existing_cloudwatch_log_group && var.attach_create_log_group_permission ? "logs:CreateLogGroup" : "",
```

Comment:
If you want to reuse the existing cloudwatch log group (from previous runs), set `use_existing_cloudwatch_log_group = true`.

I am not sure that we need to have an identical variable `attach_create_log_group_permission`.

Right?

Reply:
The scenario I am dealing with is ephemeral environments. When I first create a new environment the cloudwatch log group does not exist. So I need `use_existing_cloudwatch_log_group` set to false. When I destroy the environment, the log group may or may not persist. I then want to recreate that environment at a later date which, if the group exists, will fail. 

`use_existing_cloudwatch_log_group` only works if you know the CloudWatch group will be there, I do not.

As stated in the feature request, a work around would be to create the CloudWatch group separately. However, I would rather contain the logic to a single module.

Reply:
To be honest, since we provision the CloudWatch group separately, having `CreateLogGroup` on the Lambda seems redundant to me unless there is something I'm missing.

Reply:
@antonbabenko thoughts?